scylla/src/transport/cql_types_test.rs

```diff
@@ -116,9 +116,103 @@ async fn test_varint() {
         "-129",
         "123456789012345678901234567890",
         "-123456789012345678901234567890",
+    ]
+}
+
+#[cfg(feature = "num-bigint-03")]
+#[tokio::test]
+async fn test_varint03() {
+    let tests = varint_test_cases();
+    run_tests::<num_bigint_03::BigInt>(&tests, "varint").await;
+}
+
+#[cfg(feature = "num-bigint-04")]
+#[tokio::test]
+async fn test_varint04() {
+    let tests = varint_test_cases();
+    run_tests::<num_bigint_04::BigInt>(&tests, "varint").await;
+}
+
+#[tokio::test]
+async fn test_cql_varint() {
+    let tests = [
+        vec![0x00],       // 0
+        vec![0x01],       // 1
+        vec![0x00, 0x01], // 1 (with leading zeros)
+        vec![0x7F],       // 127
+        vec![0x00, 0x80], // 128
+        vec![0x00, 0x81], // 129
+        vec![0xFF],       // -1
+        vec![0x80],       // -128
+        vec![0xFF, 0x7F], // -129
+        vec![
+            0x01, 0x8E, 0xE9, 0x0F, 0xF6, 0xC3, 0x73, 0xE0, 0xEE, 0x4E, 0x3F, 0x0A, 0xD2,
+        ], // 123456789012345678901234567890
+        vec![
+            0xFE, 0x71, 0x16, 0xF0, 0x09, 0x3C, 0x8C, 0x1F, 0x11, 0xB1, 0xC0, 0xF5, 0x2E,
+        ], // -123456789012345678901234567890
     ];
 
-    run_tests::<BigInt>(&tests, "varint").await;
+    let table_name = "cql_varint_tests";
+    let session: Session = create_new_session_builder().build().await.unwrap();
+    let ks = unique_keyspace_name();
+
+    session
+        .query(
+            format!(
+                "CREATE KEYSPACE IF NOT EXISTS {} WITH REPLICATION = \
+            {{'class' : 'NetworkTopologyStrategy', 'replication_factor' : 1}}",
+                ks
+            ),
+            &[],
+        )
+        .await
+        .unwrap();
+    session.use_keyspace(ks, false).await.unwrap();
+
+    session
+        .query(
+            format!(
+                "CREATE TABLE IF NOT EXISTS {} (id int PRIMARY KEY, val varint)",
+                table_name
+            ),
+            &[],
+        )
+        .await
+        .unwrap();
+
+    let prepared_insert = session
+        .prepare(format!(
+            "INSERT INTO {} (id, val) VALUES (0, ?)",
+            table_name
+        ))
+        .await
+        .unwrap();
+    let prepared_select = session
+        .prepare(format!("SELECT val FROM {} WHERE id = 0", table_name))
+        .await
+        .unwrap();
+
+    for test in tests {
+        let cql_varint = CqlVarint::from_signed_bytes_be_slice(&test);
+        session
+            .execute(&prepared_insert, (&cql_varint,))
+            .await
+            .unwrap();
+
+        let read_values: Vec<CqlVarint> = session
+            .execute(&prepared_select, &[])
+            .await
+            .unwrap()
+            .rows
+            .unwrap()
+            .into_typed::<(CqlVarint,)>()
+            .map(Result::unwrap)
+            .map(|row| row.0)
+            .collect::<Vec<_>>();
+
+        assert_eq!(read_values, vec![cql_varint])
+    }
```

Comment:
Why it can't be expressed as a call to `run_tests`?

Reply:
As `run_tests` requires that the type implements `std::str::FromStr`. It's not trivial to implement it without the use of the `num_bigint` crate.

Reply:
Implementing `FromStr` for hexadecimal numbers probably wouldn't be hard, but I'm not sure if it's necessary.

Note that `num_bigint` parses the strings representing the number in decimal format.

Reply:
> As `run_tests` requires that the type implements `std::str::FromStr`. It's not trivial to implement it without the use of the `num_bigint` crate.
> 
> CQL accepts `varint` string literals represented in decimal/octal. Implementing `FromStr` for hexadecimal numbers wouldn't be hard but the hexadecimal strings provided to `run_tests` couldn't be contained as literals in queries.

I don't understand the last sentence. Why couldn't they?

Reply:
Because you can't put a hexadecimal literal for a value of type varint in CQL query:
```cql
cqlsh> CREATE TABLE ks.t (a int PRIMARY KEY, b varint);
cqlsh> INSERT INTO ks.t (a, b) VALUES (0, 0x1);
InvalidRequest: Error from server: code=2200 [Invalid query] message="Invalid HEX constant (0x1) for "b" of type varint"
```

Reply:
I just found out that you actually can, with `blobAsVarint` built-in CQL function:

```
cqlsh> create table ks.t (pk int, v varint, primary key (pk));
cqlsh> insert into ks.t (pk, v) values (2, blobasvarint(0x123421934815718293482108421905702184921834021943));
cqlsh> select pk, v from ks.t;

 pk | v
----+-----------------------------------------------------------
  2 | 446351888295288528376725140517290295205369310673010563395

(1 rows)

```

Reply:
Perhaps later we can adjust `run_tests` to take a list of (Cql?)Values instead of strings, use `CqlValueDisplayer` to print them when putting them into the query and finally compare the values fetched from the database.

Reply:
Can we do it in a separate PR? I can create an issue regarding this.

Reply:
> Can we do it in a separate PR?

Yes, that's what I meant by "later".